addons/web/controllers/action.py

```diff
@@ -36,20 +34,15 @@ def load(self, action_id, additional_context=None):
             action_type = base_action[0]['type']
             if action_type == 'ir.actions.report':
                 request.update_context(bin_size=True)
-            if additional_context:
-                request.update_context(**additional_context)
-            action = request.env[action_type].sudo().browse([action_id]).read()
-            if action:
-                value = clean_action(action[0], env=request.env)
-        return value
-
-    @route('/web/action/run', type='json', auth="user")
-    def run(self, action_id, context=None):
-        if context:
-            request.update_context(**context)
-        action = request.env['ir.actions.server'].browse([action_id])
-        result = action.run()
-        return clean_action(result, env=action.env) if result else False
+            if action_type == 'ir.actions.server':
+                action = request.env["ir.actions.server"].browse([action_id])
+                result = action.run()
+                parent_path = action.sudo().path
+                if (parent_path):
+                    result['path'] = parent_path
+                return clean_action(result, env=action.env) if result else {'type': 'ir.actions.act_window_close'}
```

Comment:
```suggestion
                return clean_action(result, env=action.env)
```
it seems like `clean_action` will return an act_window_close by default AFAICT, no?